libraries/Microsoft.Bot.Builder.Dialogs/Dialog.cs

```diff
@@ -26,6 +26,12 @@ public Dialog(string dialogId)
 
         public string Id { get; }
 
+        /// <summary>
+        /// Gets or sets the telemetry client for logging events.
+        /// </summary>
+        /// <value>The Telemetry Client logger.</value>
+        public IBotTelemetryClient Logger { get; set; } = new NullBotTelemetryClient();

```

Comment:
I can be convinced for TelemetryClient.  I explicitly put Logger here since TelemetryClient is logically the superset of logger (in future samples, Application Insights will register as the ILogger) so across languages it can be more accessible.  
Also across languages, Logger is not tied to "ILogger" and figure it's a more generic term.

Reply:
I changed to TelemetryClient!